ipn/ipnlocal/web_client.go

```diff
@@ -144,3 +149,53 @@ func (b *LocalBackend) newWebClientListener(ctx context.Context, ap netip.AddrPo
 		bo:      backoff.NewBackoff("webclient-listener", logf, 30*time.Second),
 	}
 }
+
+// newWebClientAuthURL talks to the control server to create a new auth
+// URL that can be used to validate a browser session to manage this
+// tailscaled instance via the web client.
+func (b *LocalBackend) newWebClientAuthURL(ctx context.Context, src tailcfg.NodeID) (*tailcfg.WebClientAuthResponse, error) {
+	return b.doWebClientNoiseRequest(ctx, "", src)
+}
+
+// waitWebClientAuthURL connects to the control server and blocks
+// until the associated auth URL has been completed by its user,
+// or until ctx is canceled.
+func (b *LocalBackend) waitWebClientAuthURL(ctx context.Context, id string, src tailcfg.NodeID) (*tailcfg.WebClientAuthResponse, error) {
+	return b.doWebClientNoiseRequest(ctx, id, src)
+}
+
+// doWebClientNoiseRequest handles making the "/machine/webclient"
+// noise requests to the control server for web client user auth.
```

Comment:
maybe document that this method either initializes a new session, or waits for the session to be completed, based on the presence or absence of the provided `id` value ?

Reply:
done!